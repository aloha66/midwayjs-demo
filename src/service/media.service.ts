import { Provide } from '@midwayjs/decorator';

@Provide()
export class MediaService {
  async image2pdf() {
    return {
      username: 'mockedName',
      phone: '12345678901',
      email: 'xxx.xxx@xxx.com',
    };
  }
}
