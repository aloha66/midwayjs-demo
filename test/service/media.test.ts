import { createApp, close } from '@midwayjs/mock'
import { Framework } from '@midwayjs/koa'
import { MediaService } from '@/service'

describe('test/service/media.test.ts', () => {

    it('should GET /', async () => {
    // create app
        const app = await createApp<Framework>()

        // 传入 class 忽略泛型也能正确推导
        const mediaService = await app.getApplicationContext().getAsync(MediaService)
        const ee = await mediaService.image2pdf()

        expect(ee).toMatchSnapshot()
   
        // close app
        await close(app)
    })

})
