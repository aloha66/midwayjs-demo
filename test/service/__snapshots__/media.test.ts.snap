// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/service/media.test.ts should GET / 1`] = `
{
  "email": "xxx.xxx@xxx.com",
  "phone": "12345678901",
  "username": "mockedName",
}
`;
